core/src/rust/tantivy_utils/src/query/prefix_query.rs

```diff
@@ -0,0 +1,96 @@
+//! Query that does a prefix match
```

Comment:
Perhaps you,  @alextheimer and I can sync up and understand more about this piece of code.